### PR TITLE
HWKAPM-546 Enable filtering on URI and operation in criteria

### PR DESCRIPTION
**Pull request by objectiser**

### CODE REVIEW

**Comment by objectiser** [comment]
@pavolloffay Could you take a look at this one sometime today, as needed for some UI work? Thanks.

File: server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/ElasticsearchUtil.java [hunk 1]
```diff
@@ -135,6 +138,16 @@ public static BoolQueryBuilder buildQuery(Criteria criteria, String timeProperty
             query = query.must(rangeQuery);
         }
 
+        // Querying uri and operation are only relevant to NodeDetails and CompletionTime
+        if (targetClass == NodeDetails.class || targetClass == CompletionTime.class) {
+            if (criteria.getUri() != null && !criteria.getUri().trim().isEmpty()) {
```
Comment (pavolloffay) [inline server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/ElasticsearchUtil.java]: Isn't it better to have null and empty check in criteria object?

**Comment by pavolloffay** [comment]
LGTM :+1:

File: server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/ElasticsearchUtil.java (hunk 1 above)
Comment (objectiser) [inline server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/ElasticsearchUtil.java]: Possibly, but will leave for now as consistent with other properties.

**Comment by objectiser** [comment]
@pavolloffay Thanks!

**Pull request merged**